### PR TITLE
Add py-rasterio 1.0.24

### PR DESCRIPTION
**Pull request by adamjstewart**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-rasterio/package.py [hunk 1]
```diff
@@ -14,25 +14,30 @@ class PyRasterio(PythonPackage):
...
-    version('1.0a12', 'e078ca02b3513b65a9be5bb3f528b4da')
+    maintainers = ['adamjstewart']
+    import_modules = ['rasterio', 'rasterio.rio']
 
-    variant('aws', default=False,
-        description='Enable testing with Amazon Web Services')
+    version('1.0.24', sha256='4839479621045211f66868ec49625979693450bc2e476f23e7e8ac4804eaf452')
+    version('1.0a12', sha256='47d460326e04c64590ff56952271a184a6307f814efc34fb319c12e690585f3c')
 
+    depends_on('python@3:', type=('build', 'run'), when='@1.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rasterio/package.py]: https://github.com/mapbox/rasterio/issues/1613

File: var/spack/repos/builtin/packages/py-rasterio/package.py [hunk 2]
```diff
@@ -14,25 +14,30 @@ class PyRasterio(PythonPackage):
...
     homepage = "https://github.com/mapbox/rasterio"
-    url      = "https://github.com/mapbox/rasterio/archive/1.0a12.zip"
+    url      = "https://pypi.io/packages/source/r/rasterio/rasterio-1.0.24.tar.gz"
 
-    version('1.0a12', 'e078ca02b3513b65a9be5bb3f528b4da')
+    maintainers = ['adamjstewart']
+    import_modules = ['rasterio', 'rasterio.rio']
 
-    variant('aws', default=False,
-        description='Enable testing with Amazon Web Services')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-rasterio/package.py]: Will removing AWS support negatively affect somebody?  (i.e. whoever put it in..)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-rasterio/package.py]: Idk, you tell me (#7156). As far as I can tell, the AWS dependency is only needed for testing purposes, it does not provide additional functionality. The dependency is still there, but is only activated at test time.

Comment (citibeth) [reply var/spack/repos/builtin/packages/py-rasterio/package.py]: Well, that's embarassing.  I think I remember... a friend needed it, so I made the package.  I've never used it myself.  And I have no idea why there's a variant that had no effect.

**Pull request merged**